packages/adblocker-extended-selectors/rollup.config.ts

```diff
@@ -28,22 +32,37 @@ export default [
     ],
   },
   {
-    input: './dist/src/adblocker.js',
+    input: './adblocker.ts',
     output: [
       {
-        dir: './dist/esm',
+        dir: './dist',
         format: 'esm',
         preserveModules: true,
         entryFileNames: '[name].js',
         sourcemap: true,
       },
       {
-        dir: './dist/cjs',
+        dir: './dist',
         format: 'cjs',
         preserveModules: true,
         entryFileNames: '[name].cjs',
         sourcemap: true,
       },
     ],
+    plugins: [
+      // compilerOptions are here a workaround for @rollup/plugin-typescript not being able to emit declarations
+      typescript({ compilerOptions: { declarationDir: './dist/types' } }),
+    ],
   },
+  {
+    input: 'dist/types/adblocker.d.ts',
+    output: [
+      { file: `dist/adblocker.d.cts` },
+      { file: `dist/adblocker.d.mts` },
+      { file: `dist/adblocker.d.ts` },
```

Comment:
We may export only two: `d.cts` and `d.ts` (equivalent of `d.mts` for our libraries since we have `"type": "module"` in `package.json`). Also, `d.ts` extension is still required for `types` field of `package.json` for legacy support.